python3Packages.secretstorage: 2.3.1 -> 3.1.0

worldofpeace opened the pull request:
#### Changelog: https://github.com/mitya57/secretstorage/blob/master/changelog

Pertinent changes:

- ported from dbus-python to jeepney
  Thusly Python 3.5 or newer is required.
- We can use pypi
- Add pygobject3

###### Motivation for this change
This comment from @Mic92 https://github.com/NixOS/nixpkgs/pull/51269#issuecomment-445198589

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

worldofpeace committed:
python3Packages.jeepney: init at 0.4

worldofpeace committed:
python3Packages.secretstorage: 2.3.1 -> 3.1.0

Changelog: https://github.com/mitya57/secretstorage/blob/master/changelog

Pertinent changes:

- ported from dbus-python to jeepney
  Thusly Python 3.5 or newer is required.
- We can use pypi
- Add pygobject3

Mic92 commented on pkgs/development/python-modules/jeepney/default.nix:
```diff
@@ -0,0 +1,35 @@
+  pname = "jeepney";
+  version = "0.4";
+
+  disabled = !(pythonAtLeast "3.5");
```
```suggestion
  disabled = pythonOlder "3.5";
```

Mic92 commented on pkgs/development/python-modules/secretstorage/default.nix:
```diff
@@ -1,23 +1,28 @@
-    repo = "secretstorage";
-    rev = version;
-    sha256 = "1sjd2jjbxgkkxyrfwx89x0hsnn39w2cr2qkxbg1iz52znr4sqism";
+  disabled = !(pythonAtLeast "3.5");
```
```suggestion
  disabled = pythonOlder "3.5";
```

worldofpeace commented:
@GrahamcOfBorg build python3Packages.secretstorage python3Packages.keyring

dotlambda commented on pkgs/development/python-modules/jeepney/default.nix:
```diff
@@ -0,0 +1,35 @@
+  checkInputs = [
+    pytest
+    testpath
+    tornado_4
```
Where is this requirement specified?
It's really sad to see people adding different versions of python packages to `python-packages.nix` again and again. (I know it wasn't you).

worldofpeace replied:
> Where is this requirement specified?

My mistake, this should be `tornado` and a `propagatedBuildInputs`.
Seems to be a sort of integration https://gitlab.com/takluyver/jeepney/blob/master/docs/integrate.rst

> It's really sad to see people adding different versions of python packages to python-packages.nix again and again. (I know it wasn't you).

With enough frequency we should probably add a check to OfBorg for it.

worldofpeace commented:
@dotlambda I've corrected https://github.com/NixOS/nixpkgs/pull/51904#discussion_r241008381
Also noticed that jeepney didn't have a license so i added that.

So here's the things I've done to test:
* Ran `nix-review` successfully (unrelated cached failures ignored)
* Tested examples for jeepney in a running session with dbus

And did what I was here to do in the first place, which was to make `pgcli` execute successfully :+1:
